sdk/storage/azure-storage-blob/samples/walk_blob_hierarchy_async.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: walk_blob_hierarchy_async.py
+
+DESCRIPTION:
+    This example walks the containers and blobs within a storage account,
+    displaying them in a hierarchical structure and, when present, showing
+    the number of snapshots that are available per blob. This sample expects
+    that the `AZURE_STORAGE_CONNECTION_STRING` environment variable is set.
+    It SHOULD NOT be hardcoded in any code derived from this sample.
+
+USAGE: python walk_blob_hierarchy_async.py
+
+EXAMPLE OUTPUT:
+
+C: container1
+F:    folder1/
+F:       subfolder1/
+B:          test.rtf
+B:       test.rtf
+F:    folder2/
+B:       test.rtf
+B:    test.rtf (1 snapshots)
+B:    test2.rtf
+C: container2
+B:    demovid.mp4
+B:    mountain.jpg
+C: container3
+C: container4
+"""
+
+import asyncio
+import os
+import sys
+
+from azure.storage.blob.aio import BlobServiceClient
+
+# I should be able to import this directly from azure.storage.blob
+from azure.storage.blob._models import BlobPrefix
```

Comment:
@rakshith91 am I correct that I should be able to simply import this from azure.storage.blob? It seems this branch is missing this model from the __init__ file.

Reply:
The reason we didnt expose BlobPrefix at this point is because we have different implementations for sync and async. .. we will mostly add it to the namespace soon...for now, this should be the right way to go.